CHANGELOG.md

```diff
@@ -1,3 +1,12 @@
+## 0.5.0 (31/12/2015)
+
+### Changes
+
+* The dependency to Guzzle is removed. We are now using PHP-HTTP to make us independent from any transport library. 
+* We are using PSR7 requests and responses
+* Introduced HttpFactory to create clients and responses
```

Comment:
> First to put all discovery functions in one place.

While I get the point, this is not SOLID. Also, static factories should be used in very rare cases. As I can see from the implementation, apart from the client, all other methods are just proxies to other discovery calls.

> The second reson why I prefer a HttpFactory is because it make the creating of a client simpler when you use plugins.

Not sure if creating a client like this is a good idea. Static factories should be used as fallbacks. In that case, you probably don't have plugins to configure.